src/compat/_internal/normalizeForCase.ts

```diff
@@ -1,6 +1,6 @@
 export function normalizeForCase(str: string | object): string {
   if (typeof str === 'object') {
-    str = str.toString();
+    str = String(str);
```

Comment:
Does this improve performance?

Reply:
in my computer it becomes slower

Reply:
> Does this improve performance?

I changed this code for the case that  `str` is `null` or `undefined`. Before codes throws error in this case.

But we already guided that don't insert `null` and `undefiend` and I also founded this changes makes 1.75x slower. 

So I reverted! Thanks

Reply:
But Lodash don't throw error in that case. So I think `String(str)` is better way, because this function is for compatibility.

Reply:
Have you tried template strings? Is it faster?

Reply:
> Have you tried template strings? Is it faster?

<img width="789" alt="Screenshot 2024-09-12 at 10 44 22 PM" src="https://github.com/user-attachments/assets/4b8f60f4-c5ae-49a3-ae4c-d4128d4f386c">

it seems slow like `String()`

Reply:
Template literals also use complex and safe `toString`. So I think that above result make sense.

- https://tc39.es/ecma262/multipage/abstract-operations.html#sec-tostring
- https://tc39.es/ecma262/multipage/ecmascript-language-expressions.html#sec-template-literals-runtime-semantics-evaluation

Reply:
Why not we just add a null and undefined check at the top?

Reply:
Yes we can do that, but I will compare with Lodash source codes for more compatible!

Reply:
I think that we have to implement `estoolkit/compat/toString` before merging this. So i'm going to implement it at first!

Reply:
Okay!